lib/persistent-knex-model-mixin.js

```diff
@@ -28,13 +28,18 @@ module.exports = function PersistentKnexModelMixin (Model, knex, inOptions, user
   }
 
   Model.prototype.save = function * (options) {
-    const myKnex = knexOrTransaction(knex, options)
-    return myKnex.update(this.getDataPersistent()).into(Model.tableName).where('id', this.id).then()
+    const myKnex = knexOrTransaction(knex, options)(Model.tableName)
+    const data = this.getDataPersistent()
+    if (this.id) {
+      return myKnex.update(data).where('id', this.id).then()
+    } else {
+      return myKnex.insert(data).then()
+    }
   }
 
-  // Delete row based on where options
-  Model.prototype.destroy = function (selectOptions) {
-    return knex(Model.tableName).where(selectOptions).del()
+  Model.prototype.destroy = function (options) {
+    const myKnex = knexOrTransaction(knex, options)
+    return myKnex(Model.tableName).where('id', this.id).del().then()
```

Comment:
I saw it elsewhere in the code; I think it tells knex to coerce the return value into a Promise.